### PR TITLE
Various bug fixes

### LINKED ISSUE #1320: Bug in installer when during migration and seeding when wrong inputs given

### PR DESCRIPTION
**Pull request by ashu555**
Fixes
#1320 
#995 
#1318 
#1317 
#1315 
#1314 
#1312

### CODE REVIEW

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -111,9 +114,11 @@ public function getTotalSalesInCur2($allowedCurrencies2)
...
-        ->where('status', '=', 'success')
-        ->pluck('grand_total')->all();
+        $total = Invoice::leftJoin('payments', 'invoices.id', '=', 'payments.invoice_id')
+                 ->where('invoices.status', '=', 'success')
+                 ->orWhere('invoices.status', '=', 'partially paid')
+                 ->where('invoices.currency', $allowedCurrencies1)
+                 ->pluck('payments.amount')->all();
         $grandTotal = array_sum($total);
 
         return $grandTotal;
```
Comment (mverma16) [inline app/Http/Controllers/DashboardController.php:124]: getTotalSalesInCur1 and getTotalSalesInCur2 are exactly same and files code dry principle.

File: app/Http/Controllers/DashboardController.php [hunk 2]
```diff
@@ -144,10 +151,12 @@ public function getYearlySalesCur2($allowedCurrencies2)
     public function getYearlySalesCur1($allowedCurrencies1)
     {
```
Comment (mverma16) [inline app/Http/Controllers/DashboardController.php:152]: remove duplicate methods.

File: app/Http/Controllers/Installer/DatabaseController.php [hunk 3]
```diff
@@ -37,4 +29,86 @@ public function database()
...
+            'dbOutputLog' => $outputLog->fetch(),
+        ];
+    }
+
+    /**
+     * Check database type. If SQLite, then create the database file.
+     *
+     * @param \Symfony\Component\Console\Output\BufferedOutput $outputLog
+     */
+    private function sqlite(BufferedOutput $outputLog)
```
Comment (mverma16) [inline app/Http/Controllers/Installer/DatabaseController.php:103]: Reason for SqlLite database connection?

**Review by mverma16 — CHANGES_REQUESTED** [review]

**Review by SarfrazS — COMMENTED** [review]
Creating test cases for all PR's, i will continue tomorrow. once test cases completed than i will execute it.
https://docs.google.com/document/d/1HqdRicmgmGPnqVsS_AJhe4tiqufVzYAYENVT4E8JpFg/edit

File: app/Http/Controllers/DashboardController.php (hunk 1 above)
Comment (ashu555) [reply app/Http/Controllers/DashboardController.php:124]: Fixed

File: app/Http/Controllers/DashboardController.php [hunk 4]
```diff
@@ -5,6 +5,7 @@
 use App\Model\Common\Setting;
```
[the remark this answers is not in the document]
Comment (ashu555) [reply app/Http/Controllers/DashboardController.php:5]: Fixed

File: app/Http/Controllers/DashboardController.php (hunk 2 above)
Comment (ashu555) [reply app/Http/Controllers/DashboardController.php:152]: Fixed

File: app/Http/Controllers/Installer/DatabaseController.php (hunk 3 above)
Comment (ashu555) [reply app/Http/Controllers/Installer/DatabaseController.php:103]: Was part of package. Commented it for now if we need it in future

**Comment by mverma16** [comment]
Do not see any update on the PR since I last reviewed it. Have you pushed the changes?

**Review by mverma16 — APPROVED** [review]

**Review by SarfrazS — APPROVED** [review]
All the above issues are fixed. Only order inactive point checking. I am approving this pr developer is checking.

**Pull request merged**